numpy/core/function_base.py

```diff
@@ -1,11 +1,25 @@
 from __future__ import division, absolute_import, print_function
 
+import warnings
+import operator
+
 __all__ = ['logspace', 'linspace']
 
 from . import numeric as _nx
 from .numeric import result_type, NaN, shares_memory, MAY_SHARE_BOUNDS, TooHardError
 
 
+def index_deprecate(i):
```

Comment:
Should this be public? It is not exported in `__all__`, so perhaps prepending an underscode would be appropriate?